### ISSUE TITLE
Redirect after login

### ISSUE THREAD

**Issue by 4phun** [ampache/ampache#3364]
### Describe the bug

I want to open "https://_mydomain_.com/song.php?action=show_song&song_id=3723" Works perfectly when I'm already logged in. When I'm not, I get redirected to the login page, but then the home page is opened after login.

### To reproduce

1. Log in to your Ampache instance.
2. Go to any song's page (for example just click on any song in Recently Played on the home page).
3. Copy the link from your browser's address bar
4. Open a new tab in your browser, paste the link, hit enter: it works
5. Log out from Ampache, close the browser
6. Open your browser again, paste the link, hit enter: does not work because the home page is opened after login instead of the song's page

### Expected behavior
After login, I should be redirected to the requested page

## Environment

* Ampache version: 5.5.2
* Web server + version: Apache/2.4.38 (Raspbian)
* Server operating system: Raspbian GNU/Linux 10 (buster) 
* Client operating system: Ubuntu 22.04.1 LTS

### Client type

* Web

**Comment by lachlan-00** [comment]
you're right it definitely doesn't care about previous pages, should be able to send it when logon happens though.

should be simple enough to add

**Comment by lachlan-00** [comment]
referrer is respected now in ampache 8
